### ISSUE TITLE
Switch Gateway APIs to use Websockets

### ISSUE THREAD

**Issue by jkitman** [fedimint/fedimint#2293]
Our federation API uses JSON over WS, but the gateway API uses JSON over HTTPS.  To make it easier on those writing the clients / UI we should standardize on JSON over WS.

@okjodom

**Comment by elsirion** [comment]
I agree. We had this discussion before but working with protobuf/grpc has been a pita due to the poor support for properly typed message structs. With JSON RPC we can just use serde, wich has been working excellently (maybe except for some inefficiencies when dealing with binary data).

**Comment by okjodom** [comment]
Ack. Standardization of the rest webserver makes sense.

> We had this discussion before but working with protobuf/grpc has been a pita due to the poor support for properly typed message structs


Standardization would start from Gateway webserver, before we get to a refactor of the gateway grpc spec for lightning interfaces.

When we get to the grpc interfaces, an additional consideration would be if it's easier to ship JSON rpc spec for external parties to implement lightning extension services to the gateway.

**Comment by elsirion** [comment]
I wonder if that's still worth it or if we maybe want to look into Iroh for the GW API too instead. @m1sterc001guy @joschisan

**Comment by m1sterc001guy** [comment]
Personally I would prefer to stick with axum. I don't see much benefit in switching to websockets. Would that make iroh integration easier?
